src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx

```diff
@@ -83,22 +106,61 @@ class TableView extends React.Component<TableViewProps> {
    * Please read the comment on `_createColumn`
    */
   _updateColumn = (columnIndex: number, nextColumn: TableColumn<keyof TableDataRow>) => {
-    const {location, eventView, tableData} = this.props;
+    const {location, eventView, tableData, organization} = this.props;
 
     if (!tableData || !tableData.meta) {
       return;
     }
 
+    const payload = {
+      aggregation: String(nextColumn.aggregation),
+      field: String(nextColumn.field),
+      fieldname: nextColumn.name,
+    };
+
     const nextEventView = eventView.withUpdatedColumn(
       columnIndex,
-      {
-        aggregation: String(nextColumn.aggregation),
-        field: String(nextColumn.field),
-        fieldname: nextColumn.name,
-      },
+      payload,
       tableData.meta
     );
 
+    if (nextEventView !== eventView) {
+      let changed: 'some' | 'all' | 'aggregate' | 'field' | 'fieldname' = 'some';
+
+      const prevField = explodeField(eventView.fields[columnIndex]);
+      const nextField = explodeField(nextEventView.fields[columnIndex]);
+
+      const aggregationChanged = prevField.aggregation !== nextField.aggregation;
+      const fieldChanged = prevField.field !== nextField.field;
+      const fieldnameChanged = prevField.fieldname !== nextField.fieldname;
+
+      if (aggregationChanged && fieldChanged && fieldnameChanged) {
+        changed = 'all';
+      }
+
+      if (aggregationChanged && !fieldChanged && !fieldnameChanged) {
+        changed = 'aggregate';
+      }
+
+      if (!aggregationChanged && fieldChanged && !fieldnameChanged) {
+        changed = 'field';
+      }
+
+      if (!aggregationChanged && !fieldChanged && fieldnameChanged) {
+        changed = 'fieldname';
+      }
```

Comment:
I'm not sure how useful `some` would be. It would be more precise to append a string with a delimiter.

```
      let changed = [];

      if (aggregationChanged) {
        changed.push('aggregate');
      }

      if (fieldChanged) {
        changed.push('push');
      }

      if fieldnameChanged) {
        changed.push('fieldname');
      }

      changed = changed.join(';');
```

Reply:
I've modified `changed` to be a list and append `changed`, `field`, or `fieldname` as necessary.